### PR TITLE
Set fail-count and last-failure atomically

### PR DESCRIPTION
**Pull request by clumens**
I don't think the first two patches are strictly necessary, but it seems nice to keep a persistent connection and reduce the setup/teardown duplication.

### CODE REVIEW

File: daemons/controld/controld_te_events.c [hunk 1]
```diff
@@ -177,17 +186,38 @@ update_failcount(xmlNode * event, const char *event_node_uuid, int rc,
 
         /* Update the fail count, if we're not ignoring failures */
         if (!ignore_failures) {
-            attr_name = pcmk__failcount_name(rsc_id, task, interval_ms);
-            update_attrd(on_uname, attr_name, value, NULL, is_remote_node);
-            free(attr_name);
+            fail_pair = calloc(1, sizeof(pcmk__attrd_query_pair_t));
```
Comment (kgaillot) [inline daemons/controld/controld_te_events.c:189]: assert non-null, static analysis is pickier about that these days

File: daemons/controld/controld_te_events.c [hunk 2]
```diff
@@ -177,17 +186,38 @@ update_failcount(xmlNode * event, const char *event_node_uuid, int rc,
...
+        last_pair = calloc(1, sizeof(pcmk__attrd_query_pair_t));
+
+        last_name = pcmk__lastfailure_name(rsc_id, task, interval_ms);
+        last_pair->name = last_name;
+        last_pair->value = now;
+        last_pair->node = on_uname;
+
+        attrs = g_list_prepend(attrs, last_pair);
+
+        pcmk__attrd_api_update_list(NULL, attrs, NULL, NULL, NULL, opts);
```
Comment (kgaillot) [inline daemons/controld/controld_te_events.c:211]: We should reuse attrd_api, probably the best approach is to functionize the do_update block into controld_attrd.c

**Review by kgaillot — COMMENTED** [review]
nice :)

File: daemons/controld/controld_te_events.c (hunk 1 above)
Comment (clumens) [reply daemons/controld/controld_te_events.c:189]: Fixed.

**Comment by clumens** [comment]
> For testing, did you try killing pacemaker-attrd and then triggering a resource failure after it respawns, so we can be sure the reconnect logic works?
 
Failures after pacemaker-attrd work.

> Unfortunately there's no way at this point to ensure the CIB writes of the attributes are atomic, but I would guess in practice they should be written close enough together that the scheduler will see them together. Do the logs generally show that?

Yeah, they look pretty close to me:

```
Jun 28 14:43:15 cluster01 pacemaker-based     [1757] (cib_perform_op@patchset.c:523)    info: ++ /cib/status/node_state[@id='2']/transient_attributes[@id='2']/instance_attributes[@id='status-
2']:  <nvpair id="status-2-last-failure-dummy4.asyncmon_0" name="last-failure-dummy4#asyncmon_0" value="1656441795"/>
...
Jun 28 14:43:15 cluster01 pacemaker-based     [1757] (cib_perform_op@patchset.c:523)    info: ++ /cib/status/node_state[@id='2']/transient_attributes[@id='2']/instance_attributes[@id='status-
2']:  <nvpair id="status-2-fail-count-dummy4.asyncmon_0" name="fail-count-dummy4#asyncmon_0" value="1"/>
```

There's very little happening between those two lines, just more ipc reading.

**Pull request merged**